exercises/scale-generator/scale_generator_test.exs

```diff
@@ -0,0 +1,94 @@
+if !System.get_env("EXERCISM_TEST_EXAMPLES") do
+  Code.load_file("scale_generator.exs", __DIR__)
+end
+
+ExUnit.start
+ExUnit.configure exclude: :pending, trace: true
+
+defmodule ScaleGeneratorTest do
+  use ExUnit.Case
+
+  @major_scale_pattern           "MMmMMMm"
+  @minor_scale_pattern           "MmMMmMM"
+  @dorian_scale_pattern          "MmMMMmM"
+  @mixolydian_scale_pattern      "MMmMMmM"
+  @lydian_scale_pattern          "MMMmMMm"
+  @phrygian_scale_pattern        "mMMMmMM"
+  @locrian_scale_pattern         "mMMmMMM"
+  @harmonic_minor_scale_pattern  "MmMMmAm"
+  @octatonic_scale_pattern       "MmMmMmMm"
+  @hexatonic_scale_pattern       "MMMMMM"
+  @pentatonic_scale_pattern      "MMAMA"
+  @enigmatic_scale_pattern       "mAMMMmm"
+
+  #@tag :pending
+  test "C Major scale" do
+    assert ScaleGenerator.scale("C", @major_scale_pattern) == ~w(C D E F G A B)
```

Comment:
@parkerl I totally agree. You're right, they're based on the C#, F#, and Ruby ones, which just dump you into it, assuming you know enumeration, rotating lists, etc... The base description in x-common just throws in a lot of terms without explaining them, so I studied some music theory to try to understand it in the function documentation.

I'll try to go through and introduce tests like stepping up a semitone, tone, or accidental second from a given note, and before that maybe generating the 12 notes in the chromatic scale starting from a given note (defaulting to C).

I still don't understand why it drops the last note on all of them, but Wikipedia appeared to agree with dropping it, which seems to defeat the purpose of the last interval. [Another site](https://www.basicmusictheory.com/c-5th-intervals#st2) appears to include it, which makes much more sense to me, and uses "W" for whole steps and "H" for half-steps, which seems more readable to me than "M" and "m".

Actually, now that I look again, Wikipedia does include the last C from the example: https://en.wikipedia.org/wiki/Scale_(music)#Scales.2C_steps.2C_and_intervals, so perhaps it's the other implementations that are incorrect.